src/app/filters/filter-fields.component.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+  ViewEncapsulation
+} from '@angular/core';
+
+import { FilterConfig } from './filter-config';
+import { FilterEvent } from './filter-event';
+import { FilterField } from './filter-field';
+import { FilterQuery } from './filter-query';
+
+import * as _ from 'lodash';
+
+/**
+ * Component for the filter bar's filter entry components
+ */
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'pfng-filter-fields',
+  styleUrls: ['./filter-fields.component.less'],
+  templateUrl: './filter-fields.component.html'
+})
+export class FilterFieldsComponent implements OnInit {
+  @Input() config: FilterConfig;
+
+  @Output('onAdd') onAdd = new EventEmitter();
+  @Output('onFilterQueries') onFilterQueries = new EventEmitter();
+  @Output('onFieldSelect') onFieldSelect = new EventEmitter();
+
+  currentField: FilterField;
+  currentValue: string;
+  prevConfig: FilterConfig;
+  show: boolean = false;
+
+  constructor() {
+  }
+
+  // Initialization
+
+  ngOnInit(): void {
+    this.setupConfig();
+  }
+
+  ngDoCheck(): void {
+    // Do a deep compare on config
+    if (!_.isEqual(this.config, this.prevConfig)) {
+      this.setupConfig();
+    }
+  }
+
+  setupConfig(): void {
+    this.prevConfig = _.cloneDeep(this.config);
+
```

Comment:
No errors actually occur here. However, there are config properties that are undefined during rendering. 

That said, I've added the '?' operator in the HTML templates to allow the component to be rendered without the config. Doesn't look pretty, but no errors are output.